### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by ezekielgn**
In this branch, I worked on the following:

- Set up rails and connect to my local database.
- Create and run the necessary migration files for each table.
- Add Foreign keys to the columns referring to other tables.
- Make sure all columns that are foreign keys should have a corresponding index.
- The foreign keys were set up in a separate migration file.

### CODE REVIEW

File: db/migrate/20230816032645_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    create_table :posts do |t|
+      t.integer :author_id
+      t.string :title
+      t.string :text
+      t.integer :comments_counter
+      t.integer :likes_counter
+
+      t.timestamps
+    end
```
Comment (tresorsawasawa) [inline db/migrate/20230816032645_create_posts.rb:11]: - [ ] As required for the project to o correctly match the the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) give to you in the project requirements, kindly ensure that the `author_id` column is used as the foreign_key of the `users` table. This was supposed to be an `author` column but of  type ``references`` to the the table `users`. i.e.
```ruby
 create_table :posts do |t|
    t.string :title
    t.text :text
    t.integer :comments_counter
    t.integer :likes_counter
    t.timestamps

    t.references :author, foreign_key: { to_table: 'users' } 
  end
```

File: db/migrate/20230816032328_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.integer :user_id
+      t.integer :post_id
+      t.string :text
+
+      t.timestamps
+    end
```
Comment (tresorsawasawa) [inline db/migrate/20230816032328_create_comments.rb:9]: - [ ] As required for the project to correctly match the the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) give to you in the project requirements, kindly ensure that the foreign key to the `users` table is named `author_id`. I suggest you use the same approach, I mentioned in the previous comment even if it is added in a different file i.e. ``t.references :author, foreign_key: { to_table: 'users' }``

File: db/migrate/20230816032232_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.integer :user_id
```
Comment (tresorsawasawa) [inline db/migrate/20230816032232_create_likes.rb:4]: - [ ] As required for the project to correctly match the the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) give to you in the project requirements, kindly ensure that the foreign key to the `users` table is named `author_id`. I suggest you use the same approach, I mentioned in the previous comment even if it is added in a different file i.e. ``t.references :author, foreign_key: { to_table: 'users' }``

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @PabloEGonz 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230816032232_create_likes.rb (hunk 3 above)
Comment (ezekielgn) [reply db/migrate/20230816032232_create_likes.rb:4]: They were already set up as foreign keys but I used a new migration to do it.
image

**Comment by addod19** [comment]
## Hi @PabloEGonz, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the project has been merged

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**